src/fileutil.c

```diff
@@ -63,6 +63,15 @@ rs_file_open(char const *filename, char const *mode)
 	    return stdin;
     }
 
+    if (!force && is_write) {
+	if ((f = fopen(filename, "rb"))) {
+	    // File exists
+	    rs_error("File exists \"%s\", aborting!", filename);
+	    fclose(f);
+	    exit(RS_IO_ERROR);
```

Comment:
Failure to open the file for processing a few lines down does the same thing: exit(RS_IO_ERROR);

As far as I can check, there are no checks if rs_file_open() has failed and returned NULL, returning it and continuing execution will probably segfault later because of NULL file pointer.

Exiting and returning RS_IO_ERROR seems consistent with the current code.